midx.c

```diff
@@ -611,22 +584,27 @@ static struct pack_midx_entry *get_sorted_entries(struct multi_pack_index *m,
 }
 
 static size_t write_midx_pack_names(struct hashfile *f,
-				    char **pack_names,
+				    struct midx_info *info,
 				    uint32_t num_packs)
 {
 	uint32_t i;
 	unsigned char padding[MIDX_CHUNK_ALIGNMENT];
 	size_t written = 0;
 
 	for (i = 0; i < num_packs; i++) {
-		size_t writelen = strlen(pack_names[i]) + 1;
+		size_t writelen;
+
+		if (info[i].expired)
+			continue;
 
-		if (i && strcmp(pack_names[i], pack_names[i - 1]) <= 0)
+		writelen = strlen(info[i].pack_name) + 1;
+
+		if (i && strcmp(info[i].pack_name, info[i - 1].pack_name) <= 0)
 			BUG("incorrect pack-file order: %s before %s",
-			    pack_names[i - 1],
-			    pack_names[i]);
+			    info[i - 1].pack_name,
```

Comment:
since you're sorting by names on the complete list, it doesn't matter if "info[i-1]" is expired, right?

Reply:
An alternative would be to keep a "prev" index of the last written item and doing the bug check against it,
but this is fine.

Reply:
correct!